### PR TITLE
refactor: breaking CL tick to sqrt price math min spot price relaxation

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #6046

## What is the purpose of the change

This PR is a follow-up to: https://github.com/osmosis-labs/osmosis/pull/6317

It relaxes the tick-to-sqrt-price conversion to support the extended minimum range. 

## Testing and Verifying

The extended price range has been thoroughly tested in previous PRs related to https://github.com/osmosis-labs/osmosis/issues/5726 while guarding state compatibility with the simple check that is removed here.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by p0mvn** [comment]
Please only merge after https://github.com/osmosis-labs/osmosis/pull/6317 is merged

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM once #6317 is merged

**Comment by p0mvn** [comment]
I added a test skip here post-approval: [74f96ef](https://github.com/osmosis-labs/osmosis/pull/6319/commits/74f96ef30e9f28524fa3dcece289e6be79723d86)

It was in one of the future PRs anyway and I'm tracking removing that check.

Noticed that CI is being flaky from the last merge, and this should fix it.

**Pull request merged**
